tests/Unit/Adapter/ValidateTest.php

```diff
@@ -124,4 +124,31 @@ public function isUnsignedIntProvider(): array
             [false, '666invalid'],
         ];
     }
+
+    /**
+     * @param bool $expected
+     * @param string $objectClassName
+     *
+     * @dataProvider isValidObjectClassNameDataProvider
+     */
+    public function testisValidObjectClassName(bool $expected, string $objectClassName): void
+    {
+        $this->assertSame($expected, $this->validate->isValidObjectClassName($objectClassName));
+    }
+
+    public function isValidObjectClassNameDataProvider(): array
+    {
+        return [
+            [true, 'MyClassName'],
+            [true, '_MyClassName'],
+            [true, '_My_Class_Name_'],
+            [true, '_MyClassName_'],
+            [true, '__My__Class__Name__'],
+            [false, ''],
+            [false, '666'],
+            [true, '_666'],
+            [true, '_6_6_6_'],
+            [true, '__'],
```

Comment:
😅 not sure this is a good usecase

Reply:
@matks can you please provide some good use cases for example ? Or is just a [Hexakosioihexekontahexaphobia](https://www.verywellmind.com/hexakosioihexekontahexaphobia-2671858) ?

Reply:
Do we consider `__` a valid object type? 🤔

Reply:
Yes `__` is a valid object class name.

Example :  
![image](https://user-images.githubusercontent.com/1533248/170941754-6f6c7ae5-48eb-4065-a2ca-9af4a6bef9f1.png)
https://onlinephp.io?s=s7EvyCjg5QLB5JzE4mKF-HiFal4uTiAqKE3KyUxWKC5JLAFSaaV5ySWZ-XkKGak5OfkammBVnGWJRfEppbkFGuoeIGF1TWugaC0vVy3IwPh4KyuoamsA&v=7.3.33